### ISSUE TITLE
Make `build-system/app.js` compatible with `gulp --lazy_build`

### ISSUE THREAD

**Issue by rsimha** [ampproject/amphtml#24226]
`gulp --lazy_build` was recently introduced to speed up the local development experience. See #24141 for more details.

In this mode, the path modifications in `build-system/app.js` can break the lazy-building / serving of some unminified JS files. For example, requests for these files fail because the modifications to the requst urls done by `app.js` are invisible to the lazy builder.

```
/dist/ww.js (actually at /dist/ww.max.js)
/ww.js.map (actually at /dist/ww.max.js.map)
/dist/rtv/1908262352450/v0/amp-auto-lightbox-0.1.js (actually at /dist/v0/amp-auto-lightbox-0.1.max.js)
/dist/v0/amp-sticky-ad-0.1.max.js (aliased from /dist/v0/amp-sticky-ad-1.0.max.js)
```

This issue is meant to track and fix problems of this sort as they crop up.

**Comment by rsimha** [comment]
Related to #24333

**Comment by rsimha** [comment]
Lazy building of aliased extensions was fixed by #24258.
Pre-building files loaded by the core runtime will be fixed by #25155.
